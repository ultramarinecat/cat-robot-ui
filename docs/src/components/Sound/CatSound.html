<!DOCTYPE html>

<html>
<head>
  <title>CatSound.js</title>
  <meta http-equiv="content-type" content="text/html; charset=UTF-8">
  <meta name="viewport" content="width=device-width, target-densitydpi=160dpi, initial-scale=1.0; maximum-scale=1.0; user-scalable=0;">
  <link rel="stylesheet" media="all" href="../../../docco.css" />
</head>
<body>
  <div id="container">
    <div id="background"></div>
    
      <ul id="jump_to">
        <li>
          <a class="large" href="javascript:void(0);">Jump To &hellip;</a>
          <a class="small" href="javascript:void(0);">+</a>
          <div id="jump_wrapper">
          <div id="jump_page_wrapper">
            <div id="jump_page">
              
                
                <a class="source" href="../App/App.html">
                  src/components/App/App.js
                </a>
              
                
                <a class="source" href="../App/ErrorBoundary.html">
                  src/components/App/ErrorBoundary.js
                </a>
              
                
                <a class="source" href="../App/i18n.html">
                  src/components/App/i18n.js
                </a>
              
                
                <a class="source" href="../Auth/Auth.html">
                  src/components/Auth/Auth.js
                </a>
              
                
                <a class="source" href="../Auth/AuthActions.html">
                  src/components/Auth/AuthActions.js
                </a>
              
                
                <a class="source" href="../Auth/AuthReducer.html">
                  src/components/Auth/AuthReducer.js
                </a>
              
                
                <a class="source" href="../Button/Button.html">
                  src/components/Button/Button.js
                </a>
              
                
                <a class="source" href="../Cat/Cat.html">
                  src/components/Cat/Cat.js
                </a>
              
                
                <a class="source" href="../Cat/CatActions.html">
                  src/components/Cat/CatActions.js
                </a>
              
                
                <a class="source" href="../Cat/CatReducer.html">
                  src/components/Cat/CatReducer.js
                </a>
              
                
                <a class="source" href="../Cat/fulltilt.html">
                  src/components/Cat/fulltilt.js
                </a>
              
                
                <a class="source" href="../Connection/Connection.html">
                  src/components/Connection/Connection.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionActions.html">
                  src/components/Connection/ConnectionActions.js
                </a>
              
                
                <a class="source" href="../Connection/ConnectionReducer.html">
                  src/components/Connection/ConnectionReducer.js
                </a>
              
                
                <a class="source" href="../Connection/messages.html">
                  src/components/Connection/messages.js
                </a>
              
                
                <a class="source" href="../Intro/Intro.html">
                  src/components/Intro/Intro.js
                </a>
              
                
                <a class="source" href="../Intro/IntroActions.html">
                  src/components/Intro/IntroActions.js
                </a>
              
                
                <a class="source" href="../Particles/Particles.html">
                  src/components/Particles/Particles.js
                </a>
              
                
                <a class="source" href="../Pulse/Pulse.html">
                  src/components/Pulse/Pulse.js
                </a>
              
                
                <a class="source" href="../Router/Router.html">
                  src/components/Router/Router.js
                </a>
              
                
                <a class="source" href="CatSound.html">
                  src/components/Sound/CatSound.js
                </a>
              
                
                <a class="source" href="Mute.html">
                  src/components/Sound/Mute.js
                </a>
              
                
                <a class="source" href="Sound.html">
                  src/components/Sound/Sound.js
                </a>
              
                
                <a class="source" href="TurningSound.html">
                  src/components/Sound/TurningSound.js
                </a>
              
                
                <a class="source" href="../Spinner/Spinner.html">
                  src/components/Spinner/Spinner.js
                </a>
              
                
                <a class="source" href="../Typing/Typing.html">
                  src/components/Typing/Typing.js
                </a>
              
                
                <a class="source" href="../../configureStore.html">
                  src/configureStore.js
                </a>
              
                
                <a class="source" href="../../index.html">
                  src/index.js
                </a>
              
                
                <a class="source" href="../../serviceWorker.html">
                  src/serviceWorker.js
                </a>
              
                
                <a class="source" href="../../setupTests.html">
                  src/setupTests.js
                </a>
              
            </div>
          </div>
        </li>
      </ul>
    
    <ul class="sections">
        
          <li id="title">
              <div class="annotation">
                  <h1>CatSound.js</h1>
              </div>
          </li>
        
        
        
        <li id="section-1">
            <div class="annotation">
              
              <div class="pilwrap ">
                <a class="pilcrow" href="#section-1">&#182;</a>
              </div>
              <p>Cat sound</p>

            </div>
            
            <div class="content"><div class='highlight'><pre>
<span class="hljs-keyword">import</span> React, { memo, forwardRef } <span class="hljs-keyword">from</span> <span class="hljs-string">'react'</span>;

<span class="hljs-keyword">export</span> <span class="hljs-keyword">const</span> CAT_SOUND = <span class="hljs-string">`<span class="hljs-subst">${process.env.PUBLIC_URL}</span>/catSound.mp3`</span>;

<span class="hljs-keyword">export</span> <span class="hljs-keyword">const</span> CatSound = forwardRef(<span class="hljs-function">(<span class="hljs-params">props, ref</span>) =&gt;</span> {
  <span class="hljs-keyword">return</span> (
    <span class="xml"><span class="hljs-tag">&lt;<span class="hljs-name">audio</span> <span class="hljs-attr">ref</span>=<span class="hljs-string">{ref}</span> <span class="hljs-attr">data-testid</span>=<span class="hljs-string">"cat-audio"</span>&gt;</span>
      <span class="hljs-tag">&lt;<span class="hljs-name">source</span> <span class="hljs-attr">src</span>=<span class="hljs-string">{CAT_SOUND}</span> <span class="hljs-attr">preload</span>=<span class="hljs-string">"auto"</span> /&gt;</span>
    <span class="hljs-tag">&lt;/<span class="hljs-name">audio</span>&gt;</span>
  );
});

export default memo(CatSound);

</span></pre></div></div>
            
        </li>
        
    </ul>
  </div>
</body>
</html>
