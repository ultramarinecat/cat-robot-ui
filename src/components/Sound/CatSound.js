//
// Cat sound
//

import React, { memo, forwardRef } from 'react';

export const CAT_SOUND = `${process.env.PUBLIC_URL}/catSound.mp3`;

export const CatSound = forwardRef((props, ref) => {
  return (
    <audio ref={ref} data-testid="cat-audio">
      <source src={CAT_SOUND} preload="auto" />
    </audio>
  );
});

export default memo(CatSound);
