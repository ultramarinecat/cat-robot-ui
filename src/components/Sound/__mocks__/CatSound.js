import React, { forwardRef } from 'react';

export const playCat = jest.fn();
const pause = jest.fn();

const CatSound = forwardRef((props, ref) => {
  ref.current = {
    play: playCat,
    pause
  };

  return <div data-testid="cat-audio" />;
});

export default CatSound;
