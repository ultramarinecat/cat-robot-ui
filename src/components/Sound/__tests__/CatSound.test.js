import React from 'react';
import { render } from 'react-testing-library';

import { CatSound, CAT_SOUND } from '../CatSound.js';

const CAT_AUDIO = 'cat-audio';

describe('Cat sound', () => {
  it('should play a cat sound', () => {
    const { container, getByTestId } = render(<CatSound />);
    const catAudio = getByTestId(CAT_AUDIO);

    expect(catAudio).toBeInTheDocument();
    expect(catAudio.firstChild).toHaveAttribute('src', CAT_SOUND);

    expect(container.firstChild).toMatchSnapshot();
  });
});
