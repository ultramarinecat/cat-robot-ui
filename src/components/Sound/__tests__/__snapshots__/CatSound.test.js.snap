// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cat sound should play a cat sound 1`] = `
<audio
  data-testid="cat-audio"
>
  <source
    preload="auto"
    src="/catSound.mp3"
  />
</audio>
`;
